### DISCUSSION TITLE
Do we need a set of resource object oriented definition api?

### DISCUSSION THREAD

**Discussion by fregie** [apache/apisix#5035] [Ideas]
Like k8s,definition-oriented interface types that are more suitable for cloud native.
In the process of cloudification, the restful admin api has brought a lot of troubles.
What do you think guys?

**Comment by tokers** [comment]
Do you mean the Declarative configuration? And let APISIX itself watches on these files and makes the reaction (no matter the watch mechanism).

**Reply by fregie** [reply]
Yes,but not APISIX watches files,APISIX only need to watch etcd.
like `kubectl apply -f xxx.yaml`

**Reply by tokers** [reply]
Then who creates these objects to ETCD?

**Reply by fregie** [reply]
I think we shloud discuss  is declarative configuration api is needed and better first, then how to implement it for second.

**Reply by tokers** [reply]
Would you like to send an email to the maillist?

**Reply by fregie** [reply]
sure,by sending email to dev@apisix.apache.org?

**Reply by tokers** [reply]
Yep.
